tests/prefetch_related/tests.py

```diff
@@ -823,6 +833,17 @@ def test_to_attr_cached_property(self):
             with self.assertNumQueries(0):
                 self.assertEqual(person.cached_all_houses, all_houses)
 
+    def test_filter_deferred(self):
+        add_q = Query.add_q
+        with mock.patch.object(
+            Query, 'add_q', autospec=True,
+            side_effect=lambda self, q: add_q.__get__(self, Query)(q),
+        ) as mocked_add_q:
```

Comment:
I wish `mock` has a less verbose interface to count the number of calls while keeping a function around.

Reply:
+100